app/assets/javascripts/angular/controllers/mobile/messages.js

```diff
@@ -0,0 +1,197 @@
+(function() {
```

Comment:
Big fuckoff controller for Messages. I'll comment through this one instead of simply at the beginning.